Src/ILGPU/Frontend/Intrinsic/LanguageIntrinsics.cs

```diff
@@ -0,0 +1,238 @@
+                    arg.BasicValueType == BasicValueType.Float32 ||
+                    arg.BasicValueType == BasicValueType.Float64)
+                {
+                    continue;
+                }
+                throw location.GetNotSupportedException(
+                    ErrorMessages.NotSupportedInlinePTXFormatArgumentType,
+                    ptxExpression,
+                    arg.Type.ToString());
+            }
+
+            // The method parameter at position 0 is the PTX string.
+            // The method parameter at position 1 is the first argument to the PTX string.
+            var methodParams = context.Method.GetParameters();
+            var hasOutput = methodParams.Length >= 2 && methodParams[1].IsOut;
+
+            // Create the language statement
+            return context.Builder.CreateLanguageEmitPTX(
+                location,
+                expressions,
+                hasOutput,
+                ref arguments);
+        }
+
+        /// <summary>
+        /// Creates a new inline PTX instruction to the standard output stream.
+        /// </summary>
+        /// <param name="context">The current invocation context.</param>
+        private static ValueReference CreateLanguageEmitPTX(
+            ref InvocationContext context) =>
+            CreateLanguageEmitPTX(
+                GetEmitPTXExpression(ref context),
+                ref context);
+
+        /// <summary>
+        /// Resolves a PTX expression string.
+        /// </summary>
+        /// <param name="context">The current invocation context.</param>
+        /// <returns>The resolved PTX expression string.</returns>
+        private static string GetEmitPTXExpression(ref InvocationContext context)
+        {
+            var ptxExpression = context[0].ResolveAs<StringValue>();
+            return ptxExpression is null
+                ? throw context.Location.GetNotSupportedException(
+                    ErrorMessages.NotSupportedInlinePTXFormatConstant,
+                    context[0].ToString())
+                : ptxExpression.String ?? string.Empty;
+        }
+
+        /// <summary>
+        /// Parses the given PTX expression into an array of format expressions.
+        /// </summary>
+        /// <param name="ptxExpression">The PTX format expression.</param>
+        /// <param name="expressions">The array of managed format expressions.</param>
+        /// <returns>True, if all expressions could be parsed successfully.</returns>
+        public static bool TryParse(
+            string ptxExpression,
+            out FormatArray expressions)
+        {
+            expressions = FormatArray.Empty;
+
+            // Search for '%n' format arguments
+            var result = ImmutableArray.CreateBuilder<FormatExpression>(10);
+            var expression = ptxExpression.AsSpan();
+
+            while (expression.Length > 0)
+            {
+                // Search for next %
+                int foundIndex = expression.IndexOf('%');
```

Comment:
Question: May we use a `Regex` instead?

Reply:
This parsing scenario is easier than parsing `{0}`, so it could probably be done. I'll give it a try.

Reply:
Done.

Reply:
Added regex parsing of PTX expressions.